realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -208,7 +208,9 @@ public void getInstance_writeProtectedFile() throws IOException {
         assertTrue(realmFile.createNewFile());
         assertTrue(realmFile.setWritable(false));
 
-        thrown.expect(RealmIOException.class);
+        // FIXME: Why do we throw RealmIOException for this case, but IAE for other cases when opening Realm?
```

Comment:
`FileWriter` throws `IOExcetion` when opening a read-only file while `File` will throw `IllegalArgumentException`. We have probably modelled it over `FileWriter`.